app/services/service-worker.js

```diff
@@ -0,0 +1,133 @@
+import Ember from 'ember';
+
+const { Service, computed, Evented, debug } = Ember;
+
+/*
+*
+* Service worker states:
+*    "installing" - the install event has fired, but not yet complete
+*    "installed"  - install complete
+*    "activating" - the activate event has fired, but not yet complete
+*    "activated"  - fully active
+*    "redundant"  - discarded. Either failed install, or it's been
+*                   replaced by a newer version
+*  Events triggered:
+* 		registrationComplete: sw successfully registered
+*		registrationError: sw not registered
+*		newSWActive: new sw controlling page
+* 		newSWWaiting: new sw waiting for controlling page
+*		unregistrationComplete: all sw are unregistered
+*/
+
+export default Service.extend(Evented, {
+
+	sw: computed(() => window.navigator.serviceWorker).volatile(),
```

Comment:
Volatile is not necessary, this property never changes (the spec says that is [**readOnly**](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/serviceWorker))